### PR TITLE
Divide Spring unit and integration testing in codegen

### PR DESCRIPTION
**Pull request by EgorkaKulikov**
## Description

- Introduce tests type: unit and integration
- Differ codegen process for Spring unit and integration tests

### Automated tests

Verify that there are no regressions in unit testing, integration are not yet ready.

## Self-check list

- [x] I've set the proper **labels** for my PR (at least, for category and component).
- [x] PR **title** and **description** are clear and intelligible.
- [x] I've added enough **comments** to my code, particularly in hard-to-understand areas.
- [x] The functionality I've repaired, changed or added is covered with **automated tests**.
- [ ] **Manual tests** have been provided optionally.
- [x] The **documentation** for the functionality I've been working on is up-to-date.

### CODE REVIEW

File: utbot-framework/src/main/kotlin/org/utbot/framework/codegen/tree/CgAbstractSpringTestClassConstructor.kt [hunk 1]
```diff
@@ -92,17 +79,21 @@ class CgSpringTestClassConstructor(context: CgContext): CgAbstractTestClassConst
...
-        groupedModelsByClassId: Map<ClassId, Set<UtModelWrapper>>,
+    abstract fun constructClassFields(testClassModel: SpringTestClassModel): Set<CgFieldDeclaration>
+
+    abstract fun constructAdditionalMethods(): CgMethodsCluster
+
+    protected fun constructFieldsWithAnnotation(
+        groupedModelsByClassId: TypedModelWrappers,
         annotationClassId: ClassId
-    ): MutableList<CgFieldDeclaration> {
+    ): Set<CgFieldDeclaration> {
```
Comment (Damtev) [inline utbot-framework/src/main/kotlin/org/utbot/framework/codegen/tree/CgAbstractSpringTestClassConstructor.kt:89]: As I see, the class `CgFieldDeclaration` does not have an overridden method `equals`. Perhaps, we need to add it.

**Review by Damtev — CHANGES_REQUESTED** [review]

File: utbot-framework-api/src/main/kotlin/org/utbot/framework/plugin/api/Api.kt [hunk 2]
```diff
@@ -1341,19 +1341,19 @@ class SpringApplicationContext(
...
     var frameworkInstalled: Boolean = false,
 ) : CodeGenerationSettingItem {
-    UNIT_TESTS(
+    UNIT_Spring_TESTS(
         "Unit tests",
         "Unit tests",
         "Generate unit tests mocking other classes"
     ),
-    INTEGRATION_TESTS(
+    INTEGRATION_Spring_TESTS(
```
Comment (Damtev) [inline utbot-framework-api/src/main/kotlin/org/utbot/framework/plugin/api/Api.kt:1356]: It's a bit unconventional naming. To be honest, I don;t think the `Spring` part is required as it is mentioned in the name of the enum class itself.

File: utbot-framework/src/main/kotlin/org/utbot/framework/process/EngineProcessMain.kt [hunk 3]
```diff
@@ -301,4 +281,52 @@ private fun destinationWarningMessage(testPackageName: String?, classUnderTestPa
...
+            forceStaticMocking = kryoHelper.readObject(params.forceStaticMocking),
+            generateWarningsForStaticMocking = params.generateWarningsForStaticMocking,
+            runtimeExceptionTestsBehaviour = RuntimeExceptionTestsBehaviour.valueOf(params.runtimeExceptionTestsBehaviour),
+            hangingTestsTimeout = HangingTestsTimeout(params.hangingTestsTimeout),
+            enableTestsTimeout = params.enableTestsTimeout,
+            testClassPackageName = params.testClassPackageName,
+        )
+        ProjectType.Python -> error("Python code generator can not be created in Engine process")
+        ProjectType.JavaScript -> error("JavaScript code generator can not be created in Engine process")
+    }
```
Comment (Damtev) [inline utbot-framework/src/main/kotlin/org/utbot/framework/process/EngineProcessMain.kt:331]: `with(params)` looks useful here

File: utbot-framework/src/main/kotlin/org/utbot/framework/codegen/generator/CodeGenerator.kt [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import org.utbot.framework.codegen.domain.models.builders.SimpleTestClassModelBuilder
+import org.utbot.framework.codegen.services.language.CgLanguageAssistant
+import org.utbot.framework.codegen.tree.CgSimpleTestClassConstructor
+import org.utbot.framework.codegen.tree.ututils.UtilClassKind
+import org.utbot.framework.plugin.api.ClassId
+import org.utbot.framework.plugin.api.CodegenLanguage
+import org.utbot.framework.plugin.api.ExecutableId
+import org.utbot.framework.plugin.api.MockFramework
+
+open class CodeGenerator(
```
Comment (Damtev) [inline utbot-framework/src/main/kotlin/org/utbot/framework/codegen/generator/CodeGenerator.kt:20]: Perhaps, rename it to the `PureJvmCodeGenerator`, according to the corresponding `ProjectType`?

File: utbot-framework/src/main/kotlin/org/utbot/framework/process/EngineProcessMain.kt [hunk 5]
```diff
@@ -301,4 +281,52 @@ private fun destinationWarningMessage(testPackageName: String?, classUnderTestPa
...
+            projectType = ProjectType.valueOf(params.projectType),
+            generateUtilClassFile = params.generateUtilClassFile,
+            paramNames = kryoHelper.readObject(params.paramNames),
+            testFramework = testFramework,
+            mockFramework = MockFramework.valueOf(params.mockFramework),
+            codegenLanguage = CodegenLanguage.valueOf(params.codegenLanguage),
+            cgLanguageAssistant = CgLanguageAssistant.getByCodegenLanguage(CodegenLanguage.valueOf(params.codegenLanguage)),
+            parameterizedTestSource = ParametrizedTestSource.valueOf(params.parameterizedTestSource),
+            staticsMocking = staticMocking,
+            forceStaticMocking = kryoHelper.readObject(params.forceStaticMocking),
```
Comment (Damtev) [inline utbot-framework/src/main/kotlin/org/utbot/framework/process/EngineProcessMain.kt:322]: Same

File: utbot-framework/src/main/kotlin/org/utbot/framework/process/EngineProcessMain.kt [hunk 6]
```diff
@@ -301,4 +281,52 @@ private fun destinationWarningMessage(testPackageName: String?, classUnderTestPa
...
+            hangingTestsTimeout = HangingTestsTimeout(params.hangingTestsTimeout),
+            enableTestsTimeout = params.enableTestsTimeout,
+            testClassPackageName = params.testClassPackageName,
+        )
+        ProjectType.Spring -> SpringCodeGenerator(
+            springTestsType = SpringTestsType.valueOf(params.springTestsType),
+            classUnderTest = classId,
+            projectType = ProjectType.valueOf(params.projectType),
+            generateUtilClassFile = params.generateUtilClassFile,
+            paramNames = kryoHelper.readObject(params.paramNames),
```
Comment (Damtev) [inline utbot-framework/src/main/kotlin/org/utbot/framework/process/EngineProcessMain.kt:315]: It's used twice, better extract to the variable

**Review by Damtev — CHANGES_REQUESTED** [review]

File: utbot-framework/src/main/kotlin/org/utbot/framework/codegen/generator/CodeGenerator.kt (hunk 4 above)
Comment (EgorkaKulikov) [reply utbot-framework/src/main/kotlin/org/utbot/framework/codegen/generator/CodeGenerator.kt:20]: Let's do not confuse normal people about Spring :)

**Review by Damtev — APPROVED** [review]

**Pull request merged**